x-pack/legacy/plugins/apm/common/runtime_types/transaction_sample_rate_rt/index.ts

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import * as t from 'io-ts';
+
+export const transactionSampleRateRt = new t.Type<number, number, unknown>(
+  'TransactionSampleRate',
+  t.number.is,
+  (input, context) => {
+    const value = Number(input);
+    return value >= 0 && value <= 1 && Number(value.toFixed(3)) === value
```

Comment:
`Number(value.toFixed(3)) === value` looks so much nicer than the one I suggested 👍